### PR TITLE
[WIP] Rolling update improvements

### PR DESCRIPTION
**Pull request by chrislovecnm**
- improved rolling updates by cordon the node before destroy, and validating the cluster on wait
- starting the implementation of e2e testing specific to kops
- implemented create s3 bucket function

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1064)

### CODE REVIEW

File: test/ssh.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"os"
+	"io/ioutil"
+	"path/filepath"
+	"fmt"
+	"strings"
+)
+
+var (
+	// Hermione requires a PUBLIC SSH key to drop off on the cluster for access..
```
Comment (kris-nova) [inline test/ssh.go]: @chrislovecnm - this has got to go and get squashed unfortunately

File: test/ssh.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	// The private half of this key is carefully hidden in a place most users will never know about
+	DefaultSSHPublicKeyBytes = []byte("ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCnhkq2fUyDMImFAfiRItxuVNbcLc6CTj0Fipb/DWlNBhVSL34fJeUw4oFqPDuL0N2cUTsxH22drie/AOJCVzZ4f7OEECFAh6RD2I11GxqrlDppr4KjWBoV//XyGbPZa4NXg+Bpg0+UVefqjAQNiAIRmuJ8vBsl952Vkf46RIKNAznoNUI7vnYkfvU1eCCaFcWiysxfg8BALiKS4Trv00TmHO15PPuiHeYzeu4wAB+0Mtuc6uh4WmqIWOEe/jmIhdHXYniHm3AEtMjL1GVMly/QEKjANUFZCZsBFbm3S6T+o820reOXSSG1wW2XBfOSTNyGlBZJFnutQntBwvHpw6A9 kris@Goddess-Of-Production.local")
```
Comment (kris-nova) [inline test/ssh.go]: Get this key out of here please :)

File: test/testing_util.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	t, err = t.createBucket()
+	if err != nil {
+		return nil, fmt.Errorf("error in create bucket %v", err)
+	}
+	t, err = t.createClusterName()
+	if err != nil {
+		return nil, fmt.Errorf("error in create cluster name %v", err)
+	}
+
+	// TODO: setup dynamically
```
Comment (kris-nova) [inline test/testing_util.go]: RE: Dynamically generated.. it would be *fabulous* if our tests could read in a cluster spec YAML file...

**Comment by chrislovecnm** [comment]
Closing because I have done https://github.com/kubernetes/kops/pull/1134

**Pull request closed without merging**